### PR TITLE
Options: Support `#[serde(alias = name)]`

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR adds support for serde's `#[serde(alias="name")]` field-attribute to our options code generator. Simplifying supporting aliases for the same setting. 

I'm unsure if we should land this change. I may not end up using it for `line-length` because I think it's better to actually deprecate these options rather than supporting both (by adding support for Rust's `deprecated` attribute). However, it is a neat feature and it already works. 

## Test Plan

I changed `line-length` to 

```rust
    /// The line length to use when enforcing long-lines violations (like
    /// `E501`). Must be greater than `0` and less than or equal to `320`.
    #[option(
        default = "88",
        value_type = "int",
        example = r#"
        # Allow lines to be as long as 120 characters.
        line-length = 120
        "#
    )]
    #[serde(alias = "line-width", alias = "max-line-length")]
    pub line_length: Option<LineLength>,
```

And generated the options markdown

Screenshot 2023-10-18 at 08 51 25

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8000** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8000" target="_blank">Graphite</a> 
    * **PR #8002** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8002" target="_blank">Graphite</a> 
      * **PR #8006** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8006" target="_blank">Graphite</a> 
        * **PR #8031** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8031" target="_blank">Graphite</a>  👈
        * **PR #8010** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8010" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8031?utm_source=stack-comment).

File: crates/ruff_workspace/src/options.rs [hunk 1]
```diff
@@ -361,6 +361,7 @@ pub struct Options {
         line-length = 120
         "#
     )]
+    #[serde(alias = "line-width", alias = "max-line-length")]
```
Comment (charliermarsh) [inline crates/ruff_workspace/src/options.rs:364]: Is `max-line-length` for convenience, or something we once supported?

Comment (MichaReiser) [reply crates/ruff_workspace/src/options.rs]: That was a test-only change that I accidentally committed... I reverted the change

**Comment by MichaReiser** [comment]
schemars doesn't support aliases. Closing for now

**Pull request closed without merging**